packages/core/src/core/prompts.ts

```diff
@@ -128,10 +130,14 @@ export function getCoreSystemPrompt(
 - **Idiomatic Changes:** When editing, understand the local context (imports, functions/classes) to ensure your changes integrate naturally and idiomatically.
 - **Comments:** Add code comments sparingly. Focus on *why* something is done, especially for complex logic, rather than *what* is done. Only add high-value comments if necessary for clarity or if requested by the user. Do not edit comments that are separate from the code you are changing. *NEVER* talk to the user or describe your changes through comments.
 - **Proactiveness:** Fulfill the user's request thoroughly. When adding features or fixing bugs, this includes adding tests to ensure quality. Consider all created files, especially tests, to be permanent artifacts unless the user says otherwise.
-- **Confirm Ambiguity/Expansion:** Do not take significant actions beyond the clear scope of the request without confirming with the user. If asked *how* to do something, explain first, don't just do it.
+- **Handle Ambiguity/Expansion:** Do not take significant actions beyond the clear scope of the request${interactiveMode ? ` without confirming with the user. If asked *how* to do something, explain first, don't just do it` : ''}.
```

Comment:
I suggest moving ${interactiveMode ? to the beginning of this item.

```suggestion
${interactiveMode ? `**Confirm Ambiguity/Expansion:** Do not take significant actions beyond the clear scope of the request without confirming with the user. If asked *how* to do something, explain first, don't just do it.` : ''}
```